CONTRIBUTING.md

```diff
@@ -237,9 +237,83 @@ Spotless will write files to
 different copies of the formatted files, so that you can see how they
 differ and infer what is the problem.
 
-The `paddedCell()` option is disabled for normal operation in order to
-detect any misbehaviour. You can enabled the option from the command line
-by running Gradle with `-Dspotless.paddedcell`.
+The `paddedCell()` option is disabled for normal operation so that any
+misbehaviour is detected, and not just suppressed. You can enabled the
+option from the command line by running Gradle with `-Dspotless.paddedcell`.
+
+### JavaDoc
+
+Good JavaDoc can help with navigating and understanding code. Elasticsearch
+has some guidelines around when to write JavaDoc, and when not to.
```

Comment:
Javadocs are good, but they’re narrow in purpose. They are suppose to say what a method does, its parameters and return values and exceptions, and some behavior details. One concern I have is that by being prescriptive about adding Javadocs, we’re potentially going to over-index on them, and neglect adding comments in method bodies to explain the *why* of a particular implementation (due to fatigue from writing Javadocs, etc.). I’d like our guidance to be accommodating for this.